### ISSUE TITLE
[Component] [select] The remote-method is invoked after the user selects an option

### ISSUE THREAD

**Issue by Oleksii14** [element-plus/element-plus#11608]
Bug Type: **`Component`**

## Environment

- Vue Version: `3.2.47`
- Element Plus Version: `2.2.30`
- Browser / OS: `any`
- Build Tool: `Vite`

### Related Component

- `el-select`

### Reproduction Link

[Element Plus Playground](https://element-plus.run/#eyJBcHAudnVlIjoiPHNjcmlwdCBzZXR1cCBsYW5nPVwidHNcIj5cbmltcG9ydCB7IHJlZiwgdmVyc2lvbiBhcyB2dWVWZXJzaW9uIH0gZnJvbSAndnVlJ1xuaW1wb3J0IHsgdmVyc2lvbiBhcyBFcFZlcnNpb24gfSBmcm9tICdlbGVtZW50LXBsdXMnXG5pbXBvcnQgeyBFbGVtZW50UGx1cyB9IGZyb20gJ0BlbGVtZW50LXBsdXMvaWNvbnMtdnVlJ1xuXG5jb25zdCBzZWxlY3RWYWx1ZSA9IHJlZigpXG5jb25zdCByZW1vdGVNZXRob2QgPSAoKSA9PiB7XG5cdGNvbnNvbGUubG9nKCdyZW1vdGUgbWV0aG9kIGNhbGxlZCcpXG59XG48L3NjcmlwdD5cblxuPHRlbXBsYXRlPlxuICA8cD5cbiAgICA8ZWwtc2VsZWN0XG4gICAgXHR2LW1vZGVsPVwic2VsZWN0VmFsdWVcIiBcbiAgICAgIGZpbHRlcmFibGVcbiAgICAgIHJlbW90ZVxuICAgIFx0cmVzZXJ2ZS1rZXl3b3JkXG4gICAgICA6cmVtb3RlLW1ldGhvZD1cInJlbW90ZU1ldGhvZFwiXG4gICAgPlxuICAgIFx0PGVsLW9wdGlvbiBsYWJlbD1cInRlc3RcIiA6dmFsdWU9XCIxXCIgLz5cbiAgICA8L2VsLXNlbGVjdD5cbiAgICBFbGVtZW50IFBsdXMge3sgRXBWZXJzaW9uIH19ICsgVnVlIHt7IHZ1ZVZlcnNpb24gfX1cbiAgPC9wPlxuPC90ZW1wbGF0ZT5cbiIsImltcG9ydF9tYXAuanNvbiI6IntcbiAgXCJpbXBvcnRzXCI6IHt9XG59IiwiX28iOnt9fQ==)

### Steps to reproduce

- visit the reproduction example link
- open dev tools -> console
- click on select
- see options list -> select an option
- see that the `remote-method` is invoked (message in the console)

### What is Expected?

The `remote-method` should not be called after the option selection

### What is actually happening?

The `remote-method` is being called each time the user selects an option. It should be called only when user is typing

### Additional comments

_(empty)_

**Comment by acyza** [comment]
Called remote-method is used to clear filter.

**Comment by mcrio** [comment]
Hi @acyza Do you mean it's intended behavior? I'm not really sure I understand why there's a need to call the remote method after an option is selected. :|

The same behavior applies for scenario: 1. Select item; 2. Click on select to show the options again (selected item is blue); 3. Click anywhere else on the page to close the control as we don't want to select another item.  << remote method called again.

Conclusion: it seems the remote method is called each time the options dropdown closes.

**Comment by ReWWeR** [comment]
Agreed. remote-method is called each time dropdown is closed. Even when you didn't do anything inside. From my point of view, it is unexpected behavior. In the previous element version remote-method is called only when needed.

**Comment by guillaumebriday** [comment]
Hi, 

Got the same issue, do you have a workaround? 

Thanks in advance

**Comment by Oleksii14** [comment]
In my case, the workaround is the following. Just add an extra condition in the remote method to compare the current and previous search values.

```vue
<template>
  <ElSelect
    :remote-method="searchData"
  />
</template>

<script lang="ts" setup>
import { ref } from 'vue'; 

const searchQuery = ref('')

const searchData = (query: string) => {
  // On select's "blur" event, the remoteMethod
  // To avoid redundant call, we need to check whether we need to proceed.

  if (query === searchQuery.value) return;
  searchQuery.value = query;
  // fetch data logic ...
};
</script>
```

**Comment by FelipeSobral** [comment]
Was this fixed? I'm using 2.4.2 and I got the same issue

**Comment by Dsaquel** [comment]
fixed in newer versions.
